tools/Google.Cloud.Tools.VersionCompat.Tests/ClassStructCtorTest.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace Google.Cloud.Tools.VersionCompat.Tests.ClassStructCtor
+{
+    // All class ctors are explicitly coded for clarity.
+
+    namespace ClassCtorAdded.A { public class C { public C() { } } }
+    namespace ClassCtorAdded.B { public class C { public C() { } public C(int i) { } } }
+
+    namespace ClassCtorRemoved.A { public class C { public C() { } public C(int i) { } } }
+    namespace ClassCtorRemoved.B { public class C { public C() { } } }
+
+    namespace ClassCtorMadeExported.A { public class C { private C() { } } }
+    namespace ClassCtorMadeExported.B { public class C { protected C() { } } }
+
+    namespace ClassCtorMadeNotExported.A { public class C { protected C() { } } }
+    namespace ClassCtorMadeNotExported.B { public class C { private C() { } } }
+
+    namespace ClassCtorParamNameChanged.A { public class C { public C(int i) { } } }
+    namespace ClassCtorParamNameChanged.B { public class C { public C(int j) { } } }
+
+    namespace ClassCtorParamInOutChanged1.A { public class C { public C(in string s) { } } }
+    namespace ClassCtorParamInOutChanged1.B { public class C { public C(ref string s) { } } }
+
+    namespace ClassCtorParamInOutChanged2.A { public class C { public C(ref string s) { } } }
+    namespace ClassCtorParamInOutChanged2.B { public class C { public C(out string s) => s = ""; } }
+
+    namespace ClassCtorAccessModifierChanged1.A { public class C { public C() { } } }
+    namespace ClassCtorAccessModifierChanged1.B { public class C { protected C() { } } }
+
+    namespace ClassCtorAccessModifierChanged2.A { public class C { protected C() { } } }
```

Comment:
Possibly another one from internal to private, that is a no-op?

Reply:
Done